manifests/resource/snippet.pp

```diff
@@ -0,0 +1,40 @@
+# define: nginx::resource::snippet
+#
+# This definition creates a reusable config snippet that can be included by other resources
+#
+# Parameters:
+#   [*ensure*]      - Enables or disables the specified snippet (present|absent)
+#   [*owner*]       - Defines owner of the .conf file
+#   [*group*]       - Defines group of the .conf file
+#   [*mode*]        - Defines mode of the .conf file
+#   [*raw_content*] - Raw content that will be inserted into the snipped as-is
+#
+define nginx::resource::snippet (
+  String[1] $raw_content,
+  Enum['absent', 'present'] $ensure = 'present',
+  String $owner                     = $nginx::global_owner,
+  String $group                     = $nginx::global_group,
+  Stdlib::Filemode $mode            = $nginx::global_mode,
+) {
+  if ! defined(Class['nginx']) {
+    fail('You must include the nginx base class before using any defined resources')
+  }
+
+  $name_sanitized = regsubst($name, ' ', '_', 'G')
+  $config_file = "${nginx::snippets_dir}/${name_sanitized}.conf"
+
+  concat { $config_file:
```

Comment:
Probably need a check in here to ensure that `$config_file` isn't already defined as a concat resource. If it is and you try to add another snippet to the same `$config_file`, it will fail when it tries to create this resource.

Reply:
I'm not sure about this. $config_file contains the name of the defined resource, so it's already unique in the catalog.